### PR TITLE
Improved import and BitBucket logging

### PR DESCRIPTION
**Pull request by hekonsek**
I've improved import operation and BitBucket logging. Now it is much easier to find out that there is an issue with BitBucket admin rights on the repository.

### CODE REVIEW

**Comment by rawlingsj** [comment]
unit test fails:
```
not a valid logrus Level: ""
FAIL	github.com/jenkins-x/jx/pkg/jx/cmd	34.797s
?   	github.com/jenkins-x/jx/pkg/jx/cmd/certmanager	[
```

**Comment by hekonsek** [comment]
@rawlingsj Hi James. I cannot reproduce the test error by running `make test`. Any other command I'm supposed to execute to reproduce it?

File: pkg/jx/cmd/import.go [hunk 1]
```diff
@@ -199,6 +199,8 @@ func (options *ImportOptions) addImportFlags(cmd *cobra.Command, createProject b
 
 // Run executes the command
 func (options *ImportOptions) Run() error {
+	log.ConfigureLog(options.LogLevel)
```
Comment (rawlingsj) [inline pkg/jx/cmd/import.go]: is this the error?

**Comment by rawlingsj** [comment]
@hekonsek doesn't look to be much more info TBH https://gist.github.com/rawlingsj/74ff3800d6a1b645479fb6536f441d5f

Not sure but added a comment in the review as a guess

**Comment by hekonsek** [comment]
OK, it was integration test, not unit one. Fixed.

**Comment by hekonsek** [comment]
Is this a random BDD test failure again? :) If not, then if you send me build logs, I could fix it.

**Comment by ccojocar** [comment]
@hekonsek It seems that the `test-slow-integration` fails with:
```
PR https://github.com/jstrachan/environment-production/pulls/1 has commit status success
Could not find the service URL in namespace jx-production for names myrepo, , jx-production-myrepo
panic: Test: TestCreateEnvRun Timout waiting for Terminal output: ""
goroutine 452 [running]:
github.com/jenkins-x/jx/pkg/tests.expectNoTimeoutError.func1(0x0, 0x0, 0x0, 0x0, 0x24ec520, 0xc00055f920)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/tests/helpers.go:121 +0x1e3
github.com/jenkins-x/jx/vendor/github.com/Netflix/go-expect.(*Console).Expect.func1(0xc00057a840, 0xc000eb3ea0, 0xc00066a000, 0xc000eb3eb0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/Netflix/go-expect/expect.go:72 +0x8a
github.com/jenkins-x/jx/vendor/github.com/Netflix/go-expect.(*Console).Expect(0xc00057a840, 0xc000198770, 0x1, 0x1, 0x0, 0x0, 0x24ec520, 0xc00055f920)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/Netflix/go-expect/expect.go:92 +0x85a
github.com/jenkins-x/jx/vendor/github.com/Netflix/go-expect.(*Console).ExpectString(0xc00057a840, 0x21cd524, 0x5, 0x0, 0x0, 0x4349d9, 0x22cbb78)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/Netflix/go-expect/expect.go:35 +0xac
github.com/jenkins-x/jx/pkg/jx/cmd_test.TestCreateEnvRun.func1(0xc0002dad20, 0xc00057a840)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_env_test.go:99 +0x6e
created by github.com/jenkins-x/jx/pkg/jx/cmd_test.TestCreateEnvRun
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_env_test.go:97 +0x1377
FAIL	github.com/jenkins-x/jx/pkg/jx/cmd	34.923s
```

**Comment by ccojocar** [comment]
@hekonsek Please could you rebase this PR and fix the unit test? Thanks

**Comment by hekonsek** [comment]
I got some time again, so let's try to progress with this one. Rebased and pushed.

**Comment by jstrachan** [comment]
@hekonsek thanks, will try get it merged ASAP...

**Comment by jstrachan** [comment]
@hekonsek does `make  test-slow-integration` work for you locally? We get this in the CI pipeline:
```
 make test-slow-integration
go: downloading github.com/stretchr/testify v1.2.2
go: downloading github.com/Netflix/go-expect v0.0.0-20180814212900-124a37274874
go: downloading github.com/petergtz/pegomock v0.0.0-20181008215750-9750219ad78b
go: downloading github.com/hinshun/vt10x v0.0.0-20180809195222-d55458df857c
go: downloading github.com/kr/pty v1.1.2
go: downloading github.com/bouk/monkey v1.0.0
go: downloading github.com/onsi/gomega v1.4.1
go: downloading github.com/pmezard/go-difflib v1.0.0
?   	github.com/jenkins-x/jx/cmd/ip	[no test files]
?   	github.com/jenkins-x/jx/cmd/jx	[no test files]
?   	github.com/jenkins-x/jx/cmd/jx/app	[no test files]
?   	github.com/jenkins-x/jx/pkg/addon	[no test files]
?   	github.com/jenkins-x/jx/pkg/apis/jenkins.io	[no test files]
?   	github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1	[no test files]
ok  	github.com/jenkins-x/jx/pkg/auth	0.618s	coverage: 24.2% of statements
ok  	github.com/jenkins-x/jx/pkg/binaries	0.590s	coverage: 55.6% of statements
ok  	github.com/jenkins-x/jx/pkg/builds	0.695s	coverage: 62.0% of statements
?   	github.com/jenkins-x/jx/pkg/chats	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/clientset/versioned	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/clientset/versioned/fake	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/clientset/versioned/scheme	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/clientset/versioned/typed/jenkins.io/v1	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/clientset/versioned/typed/jenkins.io/v1/fake	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/informers/externalversions	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/informers/externalversions/internalinterfaces	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/informers/externalversions/jenkins.io	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/informers/externalversions/jenkins.io/v1	[no test files]
?   	github.com/jenkins-x/jx/pkg/client/listers/jenkins.io/v1	[no test files]
?   	github.com/jenkins-x/jx/pkg/cloud/aks	[no test files]
ok  	github.com/jenkins-x/jx/pkg/cloud/amazon	0.219s	coverage: 10.9% of statements
ok  	github.com/jenkins-x/jx/pkg/cloud/gke	0.205s	coverage: 0.4% of statements
?   	github.com/jenkins-x/jx/pkg/cloud/iks	[no test files]
?   	github.com/jenkins-x/jx/pkg/cloud/oke	[no test files]
ok  	github.com/jenkins-x/jx/pkg/config	0.596s	coverage: 23.1% of statements
ok  	github.com/jenkins-x/jx/pkg/cve	0.827s	coverage: 43.0% of statements
?   	github.com/jenkins-x/jx/pkg/draft	[no test files]
?   	github.com/jenkins-x/jx/pkg/extensions	[no test files]
ok  	github.com/jenkins-x/jx/pkg/gits	65.129s	coverage: 21.8% of statements
?   	github.com/jenkins-x/jx/pkg/gits/mocks	[no test files]
ok  	github.com/jenkins-x/jx/pkg/helm	1.640s	coverage: 32.2% of statements
?   	github.com/jenkins-x/jx/pkg/helm/mocks	[no test files]
?   	github.com/jenkins-x/jx/pkg/issues	[no test files]
?   	github.com/jenkins-x/jx/pkg/jenkins	[no test files]
Downloading https://github.com/weaveworks/eksctl/releases/download/0.1.5/eksctl_linux_amd64.tar.gz to /tmp/common_install_test399198265/bin/eksctl.tar.gz...
Downloaded /tmp/common_install_test399198265/bin/eksctl.tar.gz
Job already exists in Jenkins at
Creating PR for environment staging from PipelineActivity jstrachan-myrepo-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app myrepo version 1.0.1 to namespace jx-staging
Creating fake Pull Request for env staging branch promote-myrepo-1.0.1 title myrepo to 1.0.1 message Promote myrepo to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-staging/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-staging/pulls/1
Cannot promote to Environment: production as precondition Environment: staging as no status
Found Promote PullRequest https://github.com/jstrachan/environment-staging/pulls/1 on PipelineActivity jstrachan-myrepo-master-1 for Environment staging
Cannot promote to Environment: production as precondition Environment: staging has status Running
Pipeline jstrachan-myrepo-master-1 promote Environment staging has PR https://github.com/jstrachan/environment-staging/pulls/1 with status pending
Cannot promote to Environment: production as precondition Environment: staging has status Running
PR https://github.com/jstrachan/environment-staging/pulls/1 is now merged
Cannot promote to Environment: production as precondition Environment: staging has status Running
PR https://github.com/jstrachan/environment-staging/pulls/1 has commit status success
Could not find the service URL in namespace jx-staging for names myrepo, , jx-staging-myrepo
Creating PR for environment production from PipelineActivity jstrachan-myrepo-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app myrepo version 1.0.1 to namespace jx-production
Creating fake Pull Request for env production branch promote-myrepo-1.0.1 title myrepo to 1.0.1 message Promote myrepo to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-production/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-production/pulls/1
Found Promote PullRequest https://github.com/jstrachan/environment-production/pulls/1 on PipelineActivity jstrachan-myrepo-master-1 for Environment production
PR https://github.com/jstrachan/environment-production/pulls/1 is now merged
PR https://github.com/jstrachan/environment-production/pulls/1 has commit status success
Could not find the service URL in namespace jx-production for names myrepo, , jx-production-myrepo
Creating PR for environment staging from PipelineActivity jstrachan-manual-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app manual version 1.0.1 to namespace jx-staging
Creating fake Pull Request for env staging branch promote-manual-1.0.1 title manual to 1.0.1 message Promote manual to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-staging/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-staging/pulls/1
Found Promote PullRequest https://github.com/jstrachan/environment-staging/pulls/1 on PipelineActivity jstrachan-manual-master-1 for Environment staging
Pipeline jstrachan-manual-master-1 promote Environment staging has PR https://github.com/jstrachan/environment-staging/pulls/1 with status pending
PR https://github.com/jstrachan/environment-staging/pulls/1 is now merged
PR https://github.com/jstrachan/environment-staging/pulls/1 has commit status success
Could not find the service URL in namespace jx-staging for names manual, , jx-staging-manual
Promoting to production version 1.0.1 for app manual
Promoting app manual version 1.0.1 to namespace jx-production
Creating fake Pull Request for env production branch promote-manual-1.0.1 title manual to 1.0.1 message Promote manual to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-production/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-production/pulls/1
Found Promote PullRequest https://github.com/jstrachan/environment-production/pulls/1 on PipelineActivity jstrachan-manual-master-1 for Environment production
Pipeline jstrachan-manual-master-1 promote Environment production has PR https://github.com/jstrachan/environment-production/pulls/1 with status pending
Pipeline jstrachan-manual-master-1 promote Environment production has PR https://github.com/jstrachan/environment-production/pulls/1 with status pending
PR https://github.com/jstrachan/environment-production/pulls/1 is now merged
PR https://github.com/jstrachan/environment-production/pulls/1 has commit status success
Could not find the service URL in namespace jx-production for names manual, , jx-production-manual
Creating PR for environment a from PipelineActivity jstrachan-parallelrepo-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app parallelrepo version 1.0.1 to namespace jx-a
Creating fake Pull Request for env a branch promote-parallelrepo-1.0.1 title parallelrepo to 1.0.1 message Promote parallelrepo to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-a/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-a/pulls/1
Creating PR for environment b from PipelineActivity jstrachan-parallelrepo-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app parallelrepo version 1.0.1 to namespace jx-b
Creating fake Pull Request for env b branch promote-parallelrepo-1.0.1 title parallelrepo to 1.0.1 message Promote parallelrepo to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-b/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-b/pulls/1
Cannot promote to Environment: c as precondition Environment: a as no status
Found Promote PullRequest https://github.com/jstrachan/environment-a/pulls/1 on PipelineActivity jstrachan-parallelrepo-master-1 for Environment a
Found Promote PullRequest https://github.com/jstrachan/environment-b/pulls/1 on PipelineActivity jstrachan-parallelrepo-master-1 for Environment b
Cannot promote to Environment: c as precondition Environment: a has status Running
Pipeline jstrachan-parallelrepo-master-1 promote Environment a has PR https://github.com/jstrachan/environment-a/pulls/1 with status pending
Pipeline jstrachan-parallelrepo-master-1 promote Environment b has PR https://github.com/jstrachan/environment-b/pulls/1 with status pending
Cannot promote to Environment: c as precondition Environment: a has status Running
PR https://github.com/jstrachan/environment-a/pulls/1 is now merged
Pipeline jstrachan-parallelrepo-master-1 promote Environment b has PR https://github.com/jstrachan/environment-b/pulls/1 with status pending
Cannot promote to Environment: c as precondition Environment: a has status Running
PR https://github.com/jstrachan/environment-a/pulls/1 has commit status success
Could not find the service URL in namespace jx-a for names parallelrepo, , jx-a-parallelrepo
Cannot promote to Environment: c as precondition Environment: b has status Running
PR https://github.com/jstrachan/environment-b/pulls/1 is now merged
PR https://github.com/jstrachan/environment-b/pulls/1 has commit status success
Could not find the service URL in namespace jx-b for names parallelrepo, , jx-b-parallelrepo
Creating PR for environment c from PipelineActivity jstrachan-parallelrepo-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app parallelrepo version 1.0.1 to namespace jx-c
Creating fake Pull Request for env c branch promote-parallelrepo-1.0.1 title parallelrepo to 1.0.1 message Promote parallelrepo to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-c/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-c/pulls/1
Found Promote PullRequest https://github.com/jstrachan/environment-c/pulls/1 on PipelineActivity jstrachan-parallelrepo-master-1 for Environment c
PR https://github.com/jstrachan/environment-c/pulls/1 is now merged
PR https://github.com/jstrachan/environment-c/pulls/1 has commit status success
Could not find the service URL in namespace jx-c for names parallelrepo, , jx-c-parallelrepo
Creating PR for environment staging from PipelineActivity jstrachan-myrepo-master-1 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app myrepo version 1.0.1 to namespace jx-staging
Creating fake Pull Request for env staging branch promote-myrepo-1.0.1 title myrepo to 1.0.1 message Promote myrepo to version 1.0.1 with number 1 and URL https://github.com/jstrachan/environment-staging/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-staging/pulls/1
Cannot promote to Environment: production as precondition Environment: staging as no status
Found Promote PullRequest https://github.com/jstrachan/environment-staging/pulls/1 on PipelineActivity jstrachan-myrepo-master-1 for Environment staging
Pipeline jstrachan-myrepo-master-1 promote Environment staging has PR https://github.com/jstrachan/environment-staging/pulls/1 with status pending
Creating PR for environment staging from PipelineActivity jstrachan-myrepo-master-2 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app myrepo version 1.0.2 to namespace jx-staging
Creating fake Pull Request for env staging branch promote-myrepo-1.0.2 title myrepo to 1.0.2 message Promote myrepo to version 1.0.2 with number 2 and URL https://github.com/jstrachan/environment-staging/pulls/2
Creating fake PullRequest number 2 at URL https://github.com/jstrachan/environment-staging/pulls/2
Cannot promote to Environment: production as precondition Environment: staging as no status
Found Promote PullRequest https://github.com/jstrachan/environment-staging/pulls/2 on PipelineActivity jstrachan-myrepo-master-2 for Environment staging
Pipeline jstrachan-myrepo-master-2 promote Environment staging has PR https://github.com/jstrachan/environment-staging/pulls/2 with status pending
Cannot promote to Environment: production as precondition Environment: staging has status Running
Pipeline jstrachan-myrepo-master-2 promote Environment staging has PR https://github.com/jstrachan/environment-staging/pulls/2 with status pending
Cannot promote to Environment: production as precondition Environment: staging has status Running
PR https://github.com/jstrachan/environment-staging/pulls/2 is now merged
Cannot promote to Environment: production as precondition Environment: staging has status Running
PR https://github.com/jstrachan/environment-staging/pulls/2 has commit status success
Could not find the service URL in namespace jx-staging for names myrepo, , jx-staging-myrepo
Creating PR for environment production from PipelineActivity jstrachan-myrepo-master-2 as current status is (*v1.PromoteActivityStep)(nil)
Promoting app myrepo version 1.0.2 to namespace jx-production
Creating fake Pull Request for env production branch promote-myrepo-1.0.2 title myrepo to 1.0.2 message Promote myrepo to version 1.0.2 with number 1 and URL https://github.com/jstrachan/environment-production/pulls/1
Creating fake PullRequest number 1 at URL https://github.com/jstrachan/environment-production/pulls/1
Found Promote PullRequest https://github.com/jstrachan/environment-production/pulls/1 on PipelineActivity jstrachan-myrepo-master-2 for Environment production
PR https://github.com/jstrachan/environment-production/pulls/1 is now merged
PR https://github.com/jstrachan/environment-production/pulls/1 has commit status success
Could not find the service URL in namespace jx-production for names myrepo, , jx-production-myrepo
Created environment testing
Job already exists in Jenkins at
Searching for repositories in Git server https://github.com owner petclinic-gcp includes * excludes WIP-* as user ****
Downloading Quickstart source zip from https://github.com/petclinic-gcp/spring-petclinic-vets-service/archive/master.zip with basic auth for user: ****
Generated quickstart at /tmp/test-create-quickstart324123716/myvets
Created project at /tmp/test-create-quickstart324123716/myvets

not a valid logrus Level: ""
FAIL	github.com/jenkins-x/jx/pkg/jx/cmd	36.977s
?   	github.com/jenkins-x/jx/pkg/jx/cmd/certmanager	[no test files]
?   	github.com/jenkins-x/jx/pkg/jx/cmd/mocks	[no test files]
?   	github.com/jenkins-x/jx/pkg/jx/cmd/mocks/matchers	[no test files]
?   	github.com/jenkins-x/jx/pkg/jx/cmd/templates	[no test files]
ok  	github.com/jenkins-x/jx/pkg/kube	4.316s	coverage: 27.9% of statements
?   	github.com/jenkins-x/jx/pkg/kube/mocks	[no test files]
?   	github.com/jenkins-x/jx/pkg/log	[no test files]
?   	github.com/jenkins-x/jx/pkg/maven	[no test files]
?   	github.com/jenkins-x/jx/pkg/nodes	[no test files]
?   	github.com/jenkins-x/jx/pkg/pipeline_events	[no test files]
ok  	github.com/jenkins-x/jx/pkg/prow	1.004s	coverage: 79.8% of statements
ok  	github.com/jenkins-x/jx/pkg/quickstarts	0.597s	coverage: 18.4% of statements
ok  	github.com/jenkins-x/jx/pkg/reports	0.319s	coverage: 39.7% of statements
?   	github.com/jenkins-x/jx/pkg/spring	[no test files]
?   	github.com/jenkins-x/jx/pkg/table	[no test files]
?   	github.com/jenkins-x/jx/pkg/terraform	[no test files]
?   	github.com/jenkins-x/jx/pkg/testkube	[no test files]
?   	github.com/jenkins-x/jx/pkg/tests	[no test files]
ok  	github.com/jenkins-x/jx/pkg/util	1.679s	coverage: 19.8% of statements
?   	github.com/jenkins-x/jx/pkg/util/mocks	[no test files]
ok  	github.com/jenkins-x/jx/pkg/vault	0.290s	coverage: 100.0% of statements
ok  	github.com/jenkins-x/jx/pkg/version	0.664s	coverage: 40.0% of statements
?   	github.com/jenkins-x/jx/pkg/workflow	[no test files]
Makefile:94: recipe for target 'test-slow-integration' failed
```

**Comment by hekonsek** [comment]
Let me check it out.

**Pull request merged**